pyxem/tests/signals/test_indexation_results.py

```diff
@@ -64,6 +67,35 @@ def test_TemplateMatchingResults_plot_best_results_on_signal(
     plt.close("all")
 
 
+def test_results_dict_to_crystal_map(test_library_phases, test_lib_gen):
+    library_phases_Ni = test_library_phases
```

Comment:
I would test a multi phase result as well. The scratch file I worked with to test multiple navigation dimensions and phases looked like this (feel free to take whatever suits you):

```python
from diffpy.structure import Atom, Lattice, Structure
import hyperspy.api as hs
import numpy as np

from diffsims.libraries.structure_library import StructureLibrary
from diffsims.generators.diffraction_generator import DiffractionGenerator
from diffsims.generators.library_generator import DiffractionLibraryGenerator
from pyxem.utils.indexation_utils import index_dataset_with_template_rotation
from pyxem.signals.indexation_results import results_dict_to_crystal_map

# test_library_phases
ni = Structure(
    atoms=[Atom(atype="Ni", xyz=[0, 0, 0])],
    lattice=Lattice(3, 3, 3, 90, 90, 90)
)
al = Structure(
    atoms=[Atom(atype="Al", xyz=[0, 0, 0])],
    lattice=Lattice(3, 4, 3, 90, 90, 90)
)

library_phases = StructureLibrary(
    [
        "ni",
        "al"
    ],
    [
        ni,
        al
    ],
    [
        np.array([(0, 45, 90), (0, 44, 90), (0, 54.735, 45)]),
        np.array([(45, 45, 90), (45, 90, 45), (90, 45, 0)])
    ],
)

diff_gen = DiffractionGenerator(
    accelerating_voltage=200,
    precession_angle=1,
    shape_factor_model="linear",
    minimum_intensity=0.1,
)
lib_gen = DiffractionLibraryGenerator(diff_gen)

diff_lib = lib_gen.get_diffraction_library(
    library_phases,
    calibration=0.015,
    reciprocal_radius=1.182243629714282,
    half_shape=(64, 64),
    with_direct_beam=False,
    max_excitation_error=0.07,
)

nav_shape = (4, 6)  # (x, y)
sig_shape = (128, 128)
data_test = np.zeros(nav_shape + sig_shape)

n_phases = len(library_phases.identifiers)
phase_id = np.random.randint(0, n_phases, size=np.prod(nav_shape)).reshape(nav_shape)
phase_names = list(diff_lib.keys())

for idx in np.ndindex(*nav_shape):
    i = phase_id[idx]
    test_pattern = diff_lib[phase_names[i]]["simulations"][0].get_diffraction_pattern(
        size=sig_shape[0], sigma=3
    )
    data_test[idx] = test_pattern
s_test = hs.signals.Signal2D(data_test)

result, phasedict = index_dataset_with_template_rotation(
    s_test, diff_lib, phases=phase_names, n_best=3,
)

xmap = results_dict_to_crystal_map(result, phasedict, diffraction_library=diff_lib)
print(np.allclose(xmap.phase_id.reshape(nav_shape), result["phase_index"][..., 0]))

```